owncloudDomain/src/main/java/com/owncloud/android/domain/files/usecases/RemoveFileUseCase.kt

```diff
@@ -23,16 +25,37 @@ import com.owncloud.android.domain.files.FileRepository
 import com.owncloud.android.domain.files.model.OCFile
 
 class RemoveFileUseCase(
-    private val fileRepository: FileRepository
+    private val fileRepository: FileRepository,
+    private val setLastUsageFileUseCase: SetLastUsageFileUseCase,
 ) : BaseUseCaseWithResult<Unit, RemoveFileUseCase.Params>() {
 
     override fun run(params: Params) {
 
         require(params.listOfFilesToDelete.isNotEmpty())
 
-        return fileRepository.deleteFiles(
+        val listOfFilesToDeleteOriginal = params.listOfFilesToDelete.map { it to it.isAvailableLocally }
+
+        val deleteFiles = fileRepository.deleteFiles(
             listOfFilesToDelete = params.listOfFilesToDelete,
-            removeOnlyLocalCopy = params.removeOnlyLocalCopy
+            removeOnlyLocalCopy = params.removeOnlyLocalCopy,
+        )
+
+        if (params.removeOnlyLocalCopy) {
+            listOfFilesToDeleteOriginal.forEach { (ocFile, isAvailableLocally) ->
+                setLastUsageFile(ocFile, isAvailableLocally)
+            }
+        }
```

Comment:
Do we want to update the `lastUsage` field when we remove a file? I guess if we are removing it, we don't want the last usage info anymore.
In addition, in case we wanted to keep it, why just changing it when remove only local copy? If we are removing the remote copy as well, I guess it would be even more necessary

Reply:
Removed `setLastUsageFile` in this use case, because `LastUsage` will be changed to `null` always in this case.